processing-service-core/api/models/globalConfig.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var Q = require('q');
+var repository = require('../store/globalConfigStore.js');
+var logger = require('../helpers/loggingHelper.js');
+
+
+/**
+ * Creates or updates global config based on the globalConfigParams.
+ * 
+ * @param {String} globalConfigParams The json string represents key-value pair map of global configuration.
+ * @returns {*|d.promise|Function|promise|a|h} a promise that will be resolved or rejected based on result of create/update.
+ */
+module.exports.createGlobalConfig = function(globalConfigParams){
+  return executeConfig(
+    'Created global configuration',
+    'Failed to create global configuration',
+    repository.setGlobalConfig,
+    globalConfigParams.key,
+    globalConfigParams.default,
+    globalConfigParams.description
+  );
```

Comment:
I was talking to Dermot about this, I'm guessing you wrote the functions this way to cut down on code duplication but the way that the exectuable method is called with parameters that may or may not be there stands out as odd. This would be clearer if rewritten to have these call their appropriate store function, passing the required params and then pass the promise to the execute method so that the generic logic can be performed as below;

```
module.exports.createGlobalConfig = function(globalConfigParams){
  return executeConfig(
    'Created global configuration',
    'Failed to create global configuration',
     repository.setGlobalConfig(globalConfigParams.key, globalConfigParams.default, globalConfigParams.description)
```

```
function executeConfig(successMsg, failMsg, configPromise){  
  var deferred = Q.defer();
  
  configPromise
  .then(function(result){ 
    logger.info(successMsg);
    deferred.resolve(result);
  })
  ...
```